mirrors_countme/parse.py

```diff
@@ -24,36 +24,64 @@ def pre_process(filepath: str | Path) -> Iterator[str]:
         yield tmpfile.name
 
 
-def parse_from_iterator(args, lines):
-    if args.header or args.sqlite:
-        args.writer.write_header()
+def parse_from_iterator(
+    lines,
+    *,
+    writer,
+    matcher,
+    matchmode="countme",
+    header=True,
+    sqlite=None,
+    dupcheck=True,
+    index=True,
+):
+    if header or sqlite:
+        writer.write_header()
 
     for logf in lines:
         # Make an iterator object for the matching log lines
-        match_iter = iter(args.matcher(logf))
+        match_iter = iter(matcher(logf))
 
         # TEMP WORKAROUND: filter out match items with missing values
-        if args.matchmode == "countme":
+        if matchmode == "countme":
             match_iter = (i for i in match_iter if None not in i)
 
         # Duplicate data check (for sqlite output)
-        if args.dupcheck:
+        if dupcheck:
             for item in match_iter:
-                if args.writer.has_item(item):  # if it's already in the db...
+                if writer.has_item(item):  # if it's already in the db...
                     continue  # skip to next log
 
-                args.writer.write_item(item)  # insert it into the db
+                writer.write_item(item)  # insert it into the db
             # There should be no items left, but to be safe...
             continue
 
         # Write matching items (sqlite does commit at end, or rollback on error)
-        args.writer.write_items(match_iter)
+        writer.write_items(match_iter)
 
-    if args.index:
-        args.writer.write_index()
+    if index:
+        writer.write_index()
 
 
-def parse(args=None):
+def parse(
+    *,
+    writer,
+    matcher,
+    matchmode="countme",
+    header=True,
+    sqlite=None,
+    dupcheck=True,
+    index=None,
+    progress=False,
+    logs=None,
+):
     parse_from_iterator(
-        args, ReadProgress(args.logs, display=args.progress, pre_process=pre_process)
+        ReadProgress(logs, display=progress, pre_process=pre_process),
+        writer,
+        matcher,
+        matchmode,
+        header,
+        sqlite,
+        dupcheck,
+        index,
```

Comment:
These are keyword-only arguments 😉… `test_log()` doesn’t catch this because it only tests `parse_iterator()` .